bcmp/neighbors.c

```diff
@@ -84,12 +84,13 @@ static BmErr bcmp_send_neighbor_table(void *addr) {
   @return BmErr if failed
 */
 static BmErr bcmp_process_neighbor_table_request(BcmpProcessData data) {
-  BmErr err = BmEINVAL;
+  BmErr err = BmENOTINTREC;
   BcmpNeighborTableRequest *request = (BcmpNeighborTableRequest *)data.payload;
-  if (request && ((request->target_node_id == 0) ||
-                  node_id() == request->target_node_id)) {
+
+  if (request->target_node_id == 0 || node_id() == request->target_node_id) {
```

Comment:
So we used to check for request being non-null, however, this is one of the only message processing functions that checks for that. Most of the others, including the neighbor reply below, don't check.

The `BcmpProcessData` that is passed in is coming from a `data.payload` in packet.c. I'm not seeing anywhere this is being checked for being non-null. I think we could get into some trouble if someone sent a valid bcmp header message with no payload.

Reply:
Totally — should definitely fix — could be this PR or another one.

Reply:
Oh! Did you already fix it in packet.c:429 or is there more to do?

Reply:
Yeah I added it in, but want to add a unit test for it. Also going to add a check for a NULL header.